build.sbt

```diff
@@ -64,18 +64,34 @@ lazy val core = projectMatrix
   .settings(
     moduleName := "scalafix-core",
     buildInfoSettingsForCore,
-    libraryDependencies ++= List(
-      scalameta,
-      googleDiff,
-      collectionCompat
+    scalacOptions --= (
+      if (isScala3.value)
+        Seq("-P:semanticdb:synthetics:on")
+      else Nil
     ),
+    libraryDependencies ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((3, _)) => List(
```

Comment:
`if (isScala3.value)` is more readable if you just to branch based on the major version